projects/systelab-components/src/lib/spinner/spinner-with-decinals.component.spec.ts

```diff
@@ -0,0 +1,161 @@
+
+	beforeEach(async(() => {
+		TestBed.configureTestingModule({
+			imports:      [BrowserModule,
+				BrowserAnimationsModule,
+				FormsModule,
+				DragDropModule,
+				OverlayModule,
+				TreeModule,
+				HttpClientModule],
+			declarations: [TouchspinComponent, SpinnerTestComponent]
+		}).compileComponents();
+	}));
+
+	beforeEach(() => {
+		fixture = TestBed.createComponent(SpinnerTestComponent);
+		fixture.detectChanges();
+	});
+
+	it('should instantiate', () => {
+		expect(fixture.componentInstance).toBeDefined();
+	});
+
+	it('should have an initial value', () => {
+		checkHasValue(fixture, 34.12345);
+	});
+
+	it('should have the changed value if there is a change', () => {
+		enterText(fixture, '90.56564');
+		checkHasValue(fixture, 90.56564);
+	});
+
+	it('should have the changed value if there is a change', () => {
+		enterText(fixture, '90.565641');
+		checkHasValue(fixture, 90.56564);
+	});
+
+	it('should have the changed value if there is a change', () => {
+		enterText(fixture, '90.565648');
+		checkHasValue(fixture, 90.56565);
+	});
+
+	it('should increment the value if plus button is clicked (+1.12345)', () => {
+		clickPlusButton(fixture);
+		checkHasValue(fixture, 34.12345 + 1.12345);
+	});
+
+	it('should increment the value n times if plus button is clicked (+11.12345)', () => {
+		setStep(fixture, 11.12345);
+		clickPlusButton(fixture);
+		checkHasValue(fixture, 34.12345 + 11.12345);
+	});
+
+	it('should not increment the value if is the maximum', () => {
+		enterText(fixture, '100.001');
+		clickPlusButton(fixture);
+		checkHasValue(fixture, 100.001);
+	});
+
+	it('should decrement the value if minus button is clicked (-1.12345)', () => {
+		clickMinusButton(fixture);
+		checkHasValue(fixture, 34.12345 - 1.12345);
+	});
+
+	it('should decrement the value n times if minus button is clicked (-11.12345)', () => {
+		setStep(fixture, 11.12345);
+		clickMinusButton(fixture);
+		checkHasValue(fixture, 34.12345 - 11.12345);
+	});
+
+	it('should not decrement the value if is the minimum', () => {
+		enterText(fixture, '1.00001');
+		clickMinusButton(fixture);
+		checkHasValue(fixture, 1.00001);
+	});
+
+	it('should call method change when the minus button is clicked', () => {
+		spyOn(fixture.componentInstance, 'doValueChange');
+		clickMinusButton(fixture);
+		expect(fixture.componentInstance.doValueChange).toHaveBeenCalled();
+	});
+
+	it('should call method change when the plus button is clicked', () => {
+		spyOn(fixture.componentInstance, 'doValueChange');
+		clickPlusButton(fixture);
+		expect(fixture.componentInstance.doValueChange).toHaveBeenCalled();
+	});
+
+	it('should not call method change when the value is not changed', () => {
+		enterText(fixture, '100.001');
+		spyOn(fixture.componentInstance, 'doValueChange');
+		clickPlusButton(fixture);
+		expect(fixture.componentInstance.doValueChange).not.toHaveBeenCalled();
+	});
+
+});
+
+function clickPlusButton(fixture: ComponentFixture<SpinnerTestComponent>) {
```

Comment:
It is a pity that these methods need to be repeated in both test files (from clickPlusButton to setStep). 
In this case, maybe we could put the tests for both cases (with decimals and without) in the same file to prevent this (they are testing the same component).
Another option would be to create a helper file for the tests (something like the Widget objects we use for e2e tests)